command_parse.c

```diff
@@ -1003,40 +1003,28 @@ enum CommandResult parse_mailboxes(struct Buffer *buf, struct Buffer *s,
 enum CommandResult parse_my_hdr(struct Buffer *buf, struct Buffer *s,
                                 intptr_t data, struct Buffer *err)
 {
-  struct ListNode *n = NULL;
-  size_t keylen;
-
   mutt_extract_token(buf, s, MUTT_TOKEN_SPACE | MUTT_TOKEN_QUOTE);
   char *p = strpbrk(buf->data, ": \t");
   if (!p || (*p != ':'))
   {
     mutt_buffer_strcpy(err, _("invalid header field"));
     return MUTT_CMD_WARNING;
   }
-  keylen = p - buf->data + 1;
 
-  STAILQ_FOREACH(n, &UserHeader, entries)
-  {
-    /* see if there is already a field by this name */
-    if (mutt_istrn_equal(buf->data, n->data, keylen))
-    {
-      break;
-    }
-  }
+  struct EventHeader event = { buf };
+  struct ListNode *n = find_header(&UserHeader, buf->data);
 
   if (!n)
   {
-    /* not found, allocate memory for a new node and add it to the list */
-    n = mutt_list_insert_tail(&UserHeader, NULL);
+    add_header(&UserHeader, buf);
+    notify_send(NeoMutt->notify, NT_HEADER, NT_HEADER_ADD, &event);
   }
   else
   {
-    /* found, free the existing data */
-    FREE(&n->data);
+    update_header(n, buf);
+    notify_send(NeoMutt->notify, NT_HEADER, NT_HEADER_CHANGE, &event);
   }
```

Comment:
Please swap the conditions around.
It's easier to read, conditions of the form:
```
if (something_true)
  x;
else
  y;
```

Reply:
Much cleared, changed with 710e78f